extensions/amp-list/0.1/amp-list.js

```diff
@@ -358,12 +379,13 @@ export class AmpList extends AMP.BaseElement {
 
   /**
    * @param {string} itemsExpr
+   * @return {!Promise<!JsonObject|!Array<JsonObject>>} Resolved with JSON
```

Comment:
An `Array` of `JsonObject` is itself a `JsonObject`, so I don't think you need both.

Reply:
this was just me fixing the absence of the return type and copying the return type in function batchFetchJsonFor. I will keep as is for now, to minimize the changes and reduce the scope of other possible changes that might be needed if I change it. Thanks for noting. I'll address separately in a different PR.